docs/webrtc/nanostream_webrtc_device_selection.md

```diff
@@ -0,0 +1,89 @@
+---
+id: nanostream_webrtc_device_selection
+title: Device Selection
+sidebar_label: Device Selection
+---
+
+The Webcaster API gives you the possibility to present all currently attached audio and video devices to your users.
+The API call to requrest the device list is [getDevices()](../nanostream_webrtc_api/#getdevices).<br>
+You will receive the list of devices in the [ReceivedDeviceList](../nanostream_webrtc_api/#RtcUser+event_ReceivedDeviceList) event. After that you can show this list to the user, so he can choose which devices he wants to be used for the Webcast.
+
+## Preparation
+
+We will need two `<select>` tags in our HTML document, in order to display the attached audio & video devices in your HTML document. Also we will add a button to trigger the [startPreview(previewConfig)](../nanostream_webrtc_api/#startpreviewconfig) call after we made our device selection in th UI and a video element to show the preview in.
+
+```js
+// in your html body
+<video id="video-preview" autoplay playsinline muted></video>
+
+<select id="audio-device-list"></select>
+<select id="video-device-list"></select>
+
+<button id="start-preview-button" type="button">start preview</button>
+```
+
+Also we will create a Javascript function that populates above device selection lists later on.
+
+```js
+// adds options to a select element
+var createSelectOptions = function (elementId, devices) {
+
+  var selectOptions = document.getElementById(elementId).options;
+
+  // Add all devices by id and index
+  for (var device of devices) {
```

Comment:
> var device of devices

Shouldn't it be `in`?

Reply:
no, "for...of" gives you the entries of iterables, devices is an array